File: Challenges/Coding World Cup/2023/Day 14 - NZ vs AFG - Balls and Bats.java
import java.util.*;

public class Solution {
    static int ballsAndbats(int n, int[] s, int[] m, int x, int y) {
        int balls = 0;
        int bats = 0;
        for (int i = 0; i < s.length; i++) {
            balls += s[i];
        }
        for (int i = 0; i < m.length; i++) {
            bats += m[i];
        }

        int days = 0;
        while (x > 0 && y > 0) {
            x -= balls;
            y -= bats;
            days++;
        }

        while (x > 0) {
            x -= balls;
            days++;
        }

        while (y > 0) {
            y -= bats;
            days++;
        }

        return days;
    }
}

File: Challenges/Coding World Cup/2023/Day 14 - NZ vs AFG - Balls and Bats.py
from typing import *


def ballsAndbats(n: int, s: List[int], m: List[int], x: int, y: int) -> int:
    # Write your code here.

    balls = sum(s[:])
    bats = sum(m[:])

    days = 0
    while x > 0 and y > 0:
        x -= balls
        y -= bats
        days += 1

    while x > 0:
        x -= balls
        days += 1

    while y > 0:
        y -= bats
        days += 1

    return days
